confluent_kafka/producer.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from .cimpl import Producer as _cProducer
+from .serialization import (MessageField,
+                            SerializationContext)
+
+
+class SerializingProducer(_cProducer):
+    """
+    A high level Kafka Producer with serialization capabilities.
+
+    The SerializingProducer is thread safe and sharing a single instance across
+    threads will generally be faster than having multiple instances.
+
+    The :py:func:`SerializingProducer.produce()` method is asynchronous.
+    When called it adds the message to a queue of pending messages and
+    immediately returns. This allows the Producer to batch together individual
+    messages for efficiency.
+
+    The Producer will automatically retry failed produce requests up to
+    ``message.timeout.ms``.
+
+    .. versionadded:: 1.0.0
+
+        Setting ``enable.idempotence: True`` enables Idempotent Producer which
+        provides guaranteed ordering and exactly-once producing.
+
+    .. versionadded:: 1.4.0
+
+        The Transactional Producer allows an application to send messages to
+        multiple partitions (and topics) atomically.
+
+        The ``key_serializer`` and ``value_serializer`` classes instruct the
+        producer on how to convert the message payload to bytes.
+
+    Args:
+        conf (Config): Producer configuration
+
+        conf (ClientConfig): Client configuration
+            The following configurations are supported in addition to the ones
+            described in Client Configurations(linked below).
+
+            key.serializer (Serializer, optional): The serializer for key that
+                implements Serializer
+
+            value.serializer (Serializer, optional): The serializer for value
```

Comment:
suggest "The serializer used for message values"
(no need to add that implements serializer, as you have that in the type spec before)